app/db/serialize.ts

```diff
@@ -0,0 +1,26 @@
+import SuperJSON from "superjson";
+import { SuperJSONResult } from "superjson/dist/types";
+
+export type Serialized<P> = P & {
+  _superjson?: ReturnType<typeof SuperJSON.serialize>["meta"];
+};
```

Comment:
I did not know it before, it's part of a framework called [Blitz](https://blitzjs.com/). I think it would be interesting to check if it could be interesting for us.

I actually took most of the code for serialize from the babel plugin for superjson, that automatically adds superjson to your getServerSideProps : https://github.com/blitz-js/babel-plugin-superjson-next/blob/main/src/tools.tsx. I thought it was a little too magical (and based on babel) so I preferred to have it called explicitly.

Reply:
Looks interesting, especially the part about getting rid of GraphQL 😮